### PR TITLE
feat: add ESLint rule to enforce named ILogger conventions

### LINKED ISSUE #17468: Add ESLint rule for named Loggers
### Feature Description:

New logging guidelines are coming: https://github.com/eclipse-theia/theia/pull/15359

Once in effect, it would be nice to have an ESLint rule enforcing the guideline. The rule can check that:
- `@injectable` classes do use `ILogger` instead of `console` for logging
- `ILogger` names are following the guidelines

### PR DESCRIPTION
**Pull request by ankitsharma101**
#### What it does
Resolves #17468. 

This PR introduces a new custom ESLint rule (`named-logger-check.js`) to the `private-eslint-plugin` to enforce the new logger coding guidelines. Specifically, the rule ensures that:
1. `console.*` statements are not used inside `@injectable()` classes.
2. Injected `ILogger` instances are accompanied by an `@named(...)` decorator.
3. The string passed to `@named` follows the required naming convention: `[optional-purpose]package-name:class-name#optional-suffix`.
4. Files residing in `electron-main` are explicitly excluded from this check, as the DI container is not fully available during early bootstrap.

#### How to test
1. **Automated Tests:** Run the Mocha test suite in `dev-packages/private-eslint-plugin` to verify the AST logic and regex patterns.
2. **Manual Verification:** Add a temporary `console.log("test")` inside any `@injectable` class in the frontend (e.g., `packages/core/src/browser/application-shell.ts`). Run `yarn lint packages/core` and observe the custom ESLint error `Use injected ILogger instead of console statements in @injectable classes.`

#### Follow-ups
Aligning the rest of the codebase to pass this new rule. (Note: Existing violations are currently being addressed in PR #17541). 

#### Breaking changes
- [ ] This PR introduces breaking changes and requires careful review. If yes, the breaking changes section in the [changelog](https://github.com/eclipse-theia/theia/blob/master/CHANGELOG.md) has been updated.

#### Review checklist
- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)
- [x] User-facing text is internationalized using the `nls` service (for details, please see the [Internationalization/Localization section](https://github.com/theia-ide/theia/blob/master/doc/coding-guidelines.md#internationalizationlocalization) in the Coding Guidelines)

#### Reminder for reviewers
- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: dev-packages/private-eslint-plugin/index.js [hunk 1]
```diff
@@ -21,5 +21,6 @@ exports.rules = {
     "localization-check": require('./rules/localization-check'),
     "no-src-import": require('./rules/no-src-import'),
     "runtime-import-check": require('./rules/runtime-import-check'),
-    "shared-dependencies": require('./rules/shared-dependencies')
+    "shared-dependencies": require('./rules/shared-dependencies'),
+    "named-logger-check": require('./rules/named-logger-check')
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/index.js]: Registering the rule here only exposes it under `@theia/named-logger-check`, it doesn't activate it. The rule isn't added to [`configs/errors.eslintrc.json`](https://github.com/eclipse-theia/theia/blob/201e8aea5001e76ba316499bc634ae485b97b098/configs/errors.eslintrc.json#L143-L147) (or any other config), so `npm run lint` never runs it and the "How to test" steps in the PR description won't produce an error.

File: dev-packages/private-eslint-plugin/rules/named-logger-check.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
+/** @type {import('eslint').Rule.RuleModule} */
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.js]: Missing the EPL license header and `// @ts-check`. Every other rule file has both, e.g. [annotation-check.js](https://github.com/eclipse-theia/theia/blob/201e8aea5001e76ba316499bc634ae485b97b098/dev-packages/private-eslint-plugin/rules/annotation-check.js#L1-L16). annotation-check.js also uses proper `@typedef` imports instead of the inline `/** @type {any} */` casts here, worth following the same pattern.

File: dev-packages/private-eslint-plugin/rules/named-logger-check.spec.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
+const { RuleTester } = require('eslint');
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.spec.js]: Missing the EPL license header here too.

File: dev-packages/private-eslint-plugin/rules/named-logger-check.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+    meta: {
+        type: 'problem',
+        docs: {
+            description: 'Enforce ILogger usage and naming conventions in @injectable classes',
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.js]: New rule isn't documented in the plugin [README](https://github.com/eclipse-theia/theia/blob/201e8aea5001e76ba316499bc634ae485b97b098/dev-packages/private-eslint-plugin/README.md#L18-L47), which lists every other rule. Please add a `named-logger-check` entry.

**Review by sdirix — CHANGES_REQUESTED** [review]
Looks pretty good already! However we can only activate this if the rest of the codebase is migrated in a separate PR or within this PR as otherwise we have a lot of lint errors.

**Comment by ankitsharma101** [comment]
Hi Stefan, thanks for the review! I've updated the rule with the requested license headers, added // @ts-check with proper JSDoc typedefs to satisfy type checking, and added documentation to the README. The rule remains disabled in errors.eslintrc.json as discussed to avoid build breaks during the migration. Ready for another look!

File: dev-packages/private-eslint-plugin/rules/named-logger-check.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        },
+        messages: {
+            noConsole: 'Use injected ILogger instead of console statements in @injectable classes.',
+            missingNamed: 'Injected ILoggers must use the @named decorator.',
+            invalidNameFormat: 'Logger name must follow the convention: [optional-purpose]package-name:class-name#optional-suffix',
+        }
+    },
+    create(context) {
+        const filename = context.getFilename();
+        if (filename.includes('/electron-main/')) {
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.js]: `context.getFilename()` returns backslash paths on Windows, so `includes('/electron-main/')` won't match there and the exclusion silently fails. `runtime-import-check` normalizes first with [`relativeFilePath.replace(/\\/g, '/')`](https://github.com/eclipse-theia/theia/blob/a3326786bf47e6e12172815eba3578ca4234bb23/dev-packages/private-eslint-plugin/rules/runtime-import-check.js#L80); same is needed here.

**Review by sdirix — COMMENTED** [review]
Looks good! I did not test on Windows myself, but I think there could be a windows issue. Please see my comment.

File: dev-packages/private-eslint-plugin/rules/named-logger-check.js (hunk 5 above)
Comment (ankitsharma101) [reply dev-packages/private-eslint-plugin/rules/named-logger-check.js]: Thanks for the review. I've added the .replace(/\\/g, '/') normalization to ensure the electron-main exclusion works consistently across all OS environments.

File: dev-packages/private-eslint-plugin/rules/named-logger-check.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+        messages: {
+            noConsole: 'Use injected ILogger instead of console statements in @injectable classes.',
+            missingNamed: 'Injected ILoggers must use the @named decorator.',
+            invalidNameFormat: 'Logger name must follow the convention: [optional-purpose]package-name:class-name#optional-suffix',
+        }
+    },
+    create(context) {
+        const filename = context.getFilename().replace(/\\/g, '/');
+
+        if (filename.includes('/electron-main/')) {
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.js]: We should add a comment on why this is excluded (no ILogger setup in Electron process). Also we need a few more exceptions:
   - `core/src/node/console-logger-server.ts`: the console logger itself
   - `core/src/node/logger-cli-contribution.ts`: parses the log config the logger depends on
   - `core/src/browser/messaging/ws-connection-source.ts`: the socket the frontend logger's RPC rides on
   - `plugin-ext/src/hosted/node/plugin-host-rpc.ts`: the plugin host already routes `console.*` calls and there is no ILogger here too

File: dev-packages/private-eslint-plugin/rules/named-logger-check.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+            CallExpression(node) {
+                const isInsideInjectable = injectableClassStack.some(Boolean);
+                if (isInsideInjectable) {
+                    if (
+                        node.callee &&
+                        node.callee.type === 'MemberExpression' &&
+                        node.callee.object &&
+                        node.callee.object.type === 'Identifier' &&
+                        node.callee.object.name === 'console'
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.js]: This must be specialized even more as the current rule produces a few false positives, for example this line is flagged `commands.registerCommand(ConsoleCommands.SELECT_ALL, this.newCommandHandler(console => console.selectAll()));` although it's unrelated to the real console.

We need to check whether the variable is shadowed or not and only report the issue for non-shadowed (i.e. global) variables.

**Review by sdirix — CHANGES_REQUESTED** [review]
I test ran the lint rule and had two more findings, please have a look

**Comment by ankitsharma101** [comment]
Thanks for catching both of these, Stefan.

Added a comment for the electron-main exclusion and the four files you listed 
(console-logger-server.ts, logger-cli-contribution.ts, ws-connection-source.ts, 
plugin-host-rpc.ts) .

For the shadowing issue —  Added scope resolution so it only flags the real thing now, and threw in a test using your 
exact example so it can't sneak back in later.

File: dev-packages/private-eslint-plugin/rules/named-logger-check.js [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+                                d.expression.callee &&
+                                d.expression.callee.name === 'named'
+                        );
+
+                        if (!namedDecorator) {
+                            context.report({ node: target, messageId: 'missingNamed' });
+                        } else {
+                            const namedArg = namedDecorator.expression.arguments[0];
+                            if (namedArg && namedArg.type === 'Literal' && typeof namedArg.value === 'string') {
+                                const namePattern = /^(\[[a-zA-Z0-9_.-]+\])?[a-zA-Z0-9_.-]+:[a-zA-Z0-9_.-]+(#[a-zA-Z0-9_.-]+)?$/;
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.js]: This name pattern only checks for `something:Something` it does not really check for the actual names. The class name is easy to derive. The package name could be derived from the file path.

Could be a follow up.

**Comment by ankitsharma101** [comment]
Implemented this — the rule now checks the actual class and package name, not just the shape of the string. It pulls the expected class name from the enclosing @injectable() class, and the expected package name from the file path (packages/<name>/ or dev-packages/<name>/). If the @named(...) segments don't match either one, it reports classNameMismatch or packageNameMismatch.

It skips gracefully rather than false-flagging on things like anonymous classes, files outside packages//dev-packages/, or a non-literal argument passed to @named(...).

Added tests for the mismatch cases, plus a few edge cases I wanted to rule out: dev-packages/ as a valid root (not just packages/), anonymous class expressions, a path that just happens to contain "packages" as a substring without a real /packages/ segment, non-literal decorator args, and nested classes. Also did a clean lint run (cache cleared) against @theia/core and @theia/plugin-ext with the stricter rule on — both came back clean.

File: dev-packages/private-eslint-plugin/index.js (hunk 1 above)
Comment (sdirix) [inline dev-packages/private-eslint-plugin/index.js]: The rule is registered but still not enabled. As the logger migration is now complete, please merge in `master` and enable the rule.

File: dev-packages/private-eslint-plugin/rules/named-logger-check.spec.js [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+
+const { RuleTester } = require('eslint');
+const rule = require('./named-logger-check');
+
+const ruleTester = new RuleTester({
+    parser: require.resolve('@typescript-eslint/parser'),
+    parserOptions: { ecmaVersion: 2020, sourceType: 'module' }
+});
+
+ruleTester.run('named-logger-check', rule, {
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.spec.js]: These tests never run in CI. `dev-packages/private-eslint-plugin` has no `test` script, so `lerna run test` skips the package, and `theiaext test` would not pick them up either since it globs [`./lib/**/*.*spec.js`](https://github.com/eclipse-theia/theia/blob/6c8a997feb53de8833e5c16c5ccc10a2db0ac60f/dev-packages/private-ext-scripts/package.json#L21). Please add a `"test"` script that points mocha at `rules/*.spec.js`.

File: dev-packages/private-eslint-plugin/rules/named-logger-check.js [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+
+        /**
+        * Derives the Theia package name from a normalized (forward-slash) file path, e.g.
+        * 'packages/plugin-ext/src/main/node/plugin-deployer-impl.ts' -> 'plugin-ext'.
+        * Returns undefined for paths outside packages/ or dev-packages/ (e.g. examples/,
+        * doc/), where the convention doesn't cleanly apply.
+        * @param {string} normalizedFilename
+        */
+        function derivePackageName(normalizedFilename) {
+            const match = /(?:^|\/)(?:packages|dev-packages)\/([^/]+)\//.exec(normalizedFilename);
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.js]: The directory name is not always the package name: `packages/ai-hugging-face` is [`@theia/ai-huggingface`](https://github.com/eclipse-theia/theia/blob/6c8a997feb53de8833e5c16c5ccc10a2db0ac60f/packages/ai-hugging-face/package.json#L2). Also `exec` returns the first match, so a checkout under a directory called `packages` (`~/packages/theia/packages/core/...`) resolves to the wrong name. Reading `name` from the nearest `package.json` would fix both.

File: dev-packages/private-eslint-plugin/rules/named-logger-check.spec.js [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+                    constructor(@inject(ILogger) @named('private-eslint-plugin:GoodDevClass') logger) {}
+                }
+            `,
+            filename: 'dev-packages/private-eslint-plugin/src/rules/good-dev-class.ts'
+        },
+        {
+            code: `
+                @injectable()
+                class OutsidePackagesClass {
+                                        constructor(@inject(ILogger) @named('anything-goes-here:OutsidePackagesClass') logger) {}
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.spec.js]: Nit: Indentation slipped here.

**Review by sdirix — CHANGES_REQUESTED** [review]
As the logger migration is now complete, please merge in master and then we can enable and merge this new rule.

**Comment by ankitsharma101** [comment]
Ready for another review

File: dev-packages/private-eslint-plugin/rules/named-logger-check.js [hunk 12]
```diff
@@ -0,0 +1,249 @@
...
+ */
+
+const fs = require('fs');
+const path = require('path');
+
+/**
+ * @param {string} packageJsonPath
+ * @param {{ existsSync: typeof fs.existsSync, readFileSync: typeof fs.readFileSync }} fsImpl
+ */
+function readPackageName(packageJsonPath, fsImpl) {
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.js]: `shared-dependencies.js` already walks up to the nearest `package.json`. Can be shared.

File: dev-packages/private-eslint-plugin/rules/named-logger-check.js [hunk 13]
```diff
@@ -0,0 +1,249 @@
...
+                                const namePattern = /^(?:\[[a-zA-Z0-9_.-]+\])?([a-zA-Z0-9_.-]+):([a-zA-Z0-9_.-]+)(?:#[a-zA-Z0-9_.-]+)?$/;
+                                const match = namePattern.exec(namedArg.value);
+
+                                if (!match) {
+                                    context.report({ node: namedArg, messageId: 'invalidNameFormat' });
+                                } else {
+                                    const [, actualPackageName, actualClassName] = match;
+                                    const enclosingClass = injectableClassStack[injectableClassStack.length - 1];
+                                    const expectedClassName = enclosingClass && enclosingClass.className;
+                                    const testFsImpl = context.options && context.options[0] && context.options[0].__testFsImpl;
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.js]: Since `derivePackageName` is exported and already unit tested with the fake fs below, this plumbing can go away.

File: dev-packages/private-eslint-plugin/rules/named-logger-check.js [hunk 14]
```diff
@@ -0,0 +1,249 @@
...
+            // represents environment globals), both mean this is the real global console.
+            // Any variable with a def (parameter, `const console = ...`, etc.) is a shadow.
+            return !variable || variable.defs.length === 0;
+        }
+
+        return {
+            /**
+             * @param {ClassDeclaration} node
+             */
+            ClassDeclaration(node) {
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.js]: Only `ClassDeclaration` is tracked, so a logger in a class expression will check against the wrong class

File: dev-packages/private-eslint-plugin/rules/named-logger-check.js [hunk 15]
```diff
@@ -0,0 +1,249 @@
...
+ * prefers that package's package.json `name` field (npm scope stripped) over the directory
+ * name itself, since they can differ (e.g. directory 'ai-hugging-face' vs package name
+ * 'ai-huggingface'). Falls back to the directory name if no package.json/name is found there.
+ * Returns undefined for paths with no packages/dev-packages segment, where the convention
+ * doesn't apply (examples/, doc/, etc.) — this boundary is intentional and matches segments
+ * exactly, so 'mypackages/foo/bar.ts' does not accidentally match.
+ * @param {string} normalizedFilename
+ * @param {{ existsSync: typeof fs.existsSync, readFileSync: typeof fs.readFileSync }} [fsImpl]
+ *   Test-only override; defaults to real fs.
+ */
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.js]: Can be shortened

File: dev-packages/private-eslint-plugin/rules/named-logger-check.js [hunk 16]
```diff
@@ -0,0 +1,249 @@
...
+                                d.expression.callee.type === 'Identifier' &&
+                                d.expression.callee.name === 'named'
+                        );
+
+                        if (!namedDecorator) {
+                            context.report({ node: target, messageId: 'missingNamed' });
+                        } else {
+                            const namedArg = namedDecorator.expression.arguments[0];
+                            if (namedArg && namedArg.type === 'Literal' && typeof namedArg.value === 'string') {
+                                const namePattern = /^(?:\[[a-zA-Z0-9_.-]+\])?([a-zA-Z0-9_.-]+):([a-zA-Z0-9_.-]+)(?:#[a-zA-Z0-9_.-]+)?$/;
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.js]: With the package segment now enforced, the "good" example in the coding guidelines violates this rule: `@named('remote:BackendRemoteServiceImpl')` would need `core:BackendRemoteServiceImpl` or `[remote]core:...` ([link](https://github.com/eclipse-theia/theia/blob/11fb667399ac7df64df9edc0c835ace6b0e88c2e/doc/coding-guidelines.md#L698-L705)). Please fix the doc example in this PR.

File: dev-packages/private-eslint-plugin/rules/named-logger-check.spec.js [hunk 17]
```diff
@@ -0,0 +1,285 @@
...
+
+    it('is not fooled by an earlier "packages" segment in the path', () => {
+        const fsImpl = fakePackageJsonFs({
+            [path.join('~', 'packages', 'theia', 'packages', 'core', 'package.json')]: { name: '@theia/core' }
+        });
+        const filename = path.join('~', 'packages', 'theia', 'packages', 'core', 'src', 'foo.ts').replace(/\\/g, '/');
+        assert.strictEqual(derivePackageName(filename, fsImpl), 'core');
+    });
+
+    it('returns undefined when no package.json is found', () => {
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.spec.js]: this path has no `packages` segment, so it returns undefined before any package.json lookup. The actual missing-package.json case falls back to the directory name and is untested.

File: dev-packages/private-eslint-plugin/rules/named-logger-check.spec.js [hunk 18]
```diff
@@ -0,0 +1,276 @@
...
+            filename: repoFile('dev-packages', 'private-eslint-plugin', 'rules', 'good-dev-class.ts')
+        },
+        {
+            code: `
+                @injectable()
+                class OutsidePackagesClass {
+                    constructor(@inject(ILogger) @named('anything-goes-here:OutsidePackagesClass') logger) {}
+                }
+            `,
+            filename: repoFile('examples', 'browser-only', 'src', 'browser', 'outside-packages-class.ts')
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.spec.js]: This case does not need a real `package.json`, and using an absolute repo path makes it depend on where the repo is cloned. I checked with the repo symlinked under `/tmp/decoy/packages/theia`: `derivePackageName` then returns `monorepo` for this path instead of `undefined`, and the case flips to `packageNameMismatch`. A relative path here (like the `mypackages/foo/...` case below) avoids that.

File: dev-packages/private-eslint-plugin/rules/find-package-json.js [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+        return readPackageJsonCache.get(resolved);
+    }
+    /** @type {FoundPackageJson | undefined} */
+    let packageJson = undefined;
+    const packageJsonPath = path.resolve(resolved, 'package.json');
+    if (fs.existsSync(packageJsonPath)) {
+        try {
+            packageJson = JSON.parse(fs.readFileSync(packageJsonPath, { encoding: 'utf8' }));
+            packageJson['__filename'] = packageJsonPath;
+        } catch {
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/find-package-json.js]: Swallowing the parse error changes `shared-dependencies` behaviour: a malformed `package.json` now makes `findPackageJson` keep walking up and validate against an ancestor (ultimately the monorepo root) instead of failing. On master the `JSON.parse` threw ([shared-dependencies.js#L123](https://github.com/eclipse-theia/theia/blob/11fb667399ac7df64df9edc0c835ace6b0e88c2e/dev-packages/private-eslint-plugin/rules/shared-dependencies.js#L123-L123)). Rethrowing, or at least only catching for the `derivePackageName` caller, keeps the failure visible.

File: dev-packages/private-eslint-plugin/rules/named-logger-check.js [hunk 20]
```diff
@@ -0,0 +1,234 @@
...
+    let rootIndex = -1;
+    for (let i = 0; i < segments.length - 1; i++) {
+        if (segments[i] === 'packages' || segments[i] === 'dev-packages') {
+            rootIndex = i;
+        }
+    }
+    if (rootIndex === -1) {
+        return undefined;
+    }
+    const packageJson = readPackageJson(segments.slice(0, rootIndex + 2).join('/'));
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.js]: The last-segment heuristic fixed files inside real packages, but not files outside them. With the checkout at `/tmp/pkgtest/packages/theia`, `derivePackageName('/tmp/pkgtest/packages/theia/examples/browser-only/src/foo.ts')` returns `monorepo` instead of `undefined`, so any `@named` logger under `examples/` gets a `packageNameMismatch`. The spec works around this with a deliberately relative filename instead of fixing it.

Using the new `findPackageJson(filename)` and then checking that the found package directory's parent is `packages` or `dev-packages` gets this right regardless of what the ancestors are called, and reuses the helper for both rules.

File: dev-packages/private-eslint-plugin/rules/named-logger-check.spec.js [hunk 21]
```diff
@@ -0,0 +1,350 @@
...
+            `,
+            filename: repoFile('packages', 'my-package', 'src', 'browser', 'class-expression.ts'),
+            errors: [{ messageId: 'classNameMismatch', data: { expected: 'InnerExpression', actual: 'OuterClass' } }]
+        }
+    ]
+});
+
+describe('derivePackageName', () => {
+    it('prefers the package.json name over the directory name', () => {
+        assert.strictEqual(derivePackageName(normalizedRepoFile('packages', 'ai-hugging-face', 'src', 'browser', 'foo.ts')), 'ai-huggingface');
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/named-logger-check.spec.js]: These assertions depend on the real repo layout: `packages/ai-hugging-face` and `dev-packages/private-eslint-plugin` existing with those exact names, and `packages/my-package` not existing. Renaming either package breaks this test with a fairly cryptic failure. `tempFiles` above already isolates that, so it could be used here too.

File: dev-packages/private-eslint-plugin/rules/find-package-json.js [hunk 22]
```diff
@@ -0,0 +1,151 @@
+// *****************************************************************************
+// Copyright (C) 2026 Eclipse Foundation and others.
```
Comment (sdirix) [inline dev-packages/private-eslint-plugin/rules/find-package-json.js]: The three files added by this PR are the only ones in the repo with the Eclipse Foundation as copyright holder. The convention is to name the contributing company or individual, e.g. [shared-dependencies.js](https://github.com/eclipse-theia/theia/blob/11fb667399ac7df64df9edc0c835ace6b0e88c2e/dev-packages/private-eslint-plugin/rules/shared-dependencies.js#L2-L3). A maintainer should confirm what to use here.

File: packages/plugin-ext/src/hosted/node/plugin-host-rpc.ts [hunk 23]
```diff
@@ -16,6 +16,9 @@
 
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
+// The plugin host routes console.* calls to the frontend itself and has no ILogger available.
```
Comment (sdirix) [inline packages/plugin-ext/src/hosted/node/plugin-host-rpc.ts]:
```suggestion
// The plugin host routes console.* calls itself and has no ILogger available.
```

**Review by sdirix — APPROVED** [review]
